packages/angular-material/test/boolean-control.spec.ts

```diff
@@ -96,23 +160,181 @@ describe('Boolean control', () => {
+        component.ngOnInit();
+        expect(component.data).toBe(true);
+        expect(checkboxInstance.checked).toBe(true);
+
+        mockSubStore.next({
             jsonforms: {
                 core: {
                     data: { foo: false },
                     schema,
                 }
             }
         });
-        mockSubStore2.complete();
+        mockSubStore.complete();
         fixture.detectChanges();
         expect(component.data).toBe(false);
-    }));
+        expect(checkboxInstance.checked).toBe(false);
+    });
+    it('should update with undefined value', () => {
+        const mockSubStore = MockNgRedux.getSelectorStub();
+        component.uischema = uischema;
 
-    it('should display errors', async(() => {
+        mockSubStore.next({
+            jsonforms: {
+                core: {
+                    data,
+                    schema,
+                }
+            }
+        });
+        fixture.detectChanges();
+        component.ngOnInit();
+        expect(component.data).toBe(true);
+        expect(checkboxInstance.checked).toBe(true);
+
+        mockSubStore.next({
+            jsonforms: {
+                core: {
+                    data: { foo: undefined },
+                    schema,
+                }
+            }
+        });
+        mockSubStore.complete();
+        fixture.detectChanges();
+        expect(component.data).toBe(undefined);
+        expect(checkboxInstance.checked).toBe(false);
+    });
+    it('should update with null value', () => {
+        const mockSubStore = MockNgRedux.getSelectorStub();
+        component.uischema = uischema;
+
+        mockSubStore.next({
+            jsonforms: {
+                core: {
+                    data,
+                    schema,
+                }
+            }
+        });
+        fixture.detectChanges();
+        component.ngOnInit();
+        expect(component.data).toBe(true);
+        expect(checkboxInstance.checked).toBe(true);
+
+        mockSubStore.next({
+            jsonforms: {
+                core: {
+                    data: { foo: null },
+                    schema,
+                }
+            }
+        });
+        mockSubStore.complete();
+        fixture.detectChanges();
+        expect(component.data).toBe(null);
+        expect(checkboxInstance.checked).toBe(false);
+    });
+    it('should not update with wrong ref', () => {
+        const mockSubStore = MockNgRedux.getSelectorStub();
+        component.uischema = uischema;
+
+        mockSubStore.next({
+            jsonforms: {
+                core: {
+                    data,
+                    schema,
+                }
+            }
+        });
+        fixture.detectChanges();
+        component.ngOnInit();
+        expect(component.data).toBe(true);
+        expect(checkboxInstance.checked).toBe(true);
+
+        mockSubStore.next({
+            jsonforms: {
+                core: {
+                    data: { foo: true, bar: false },
```

Comment:
I'm not sure how the test name relates to the `data` here. Mind to clarify?

Reply:
the idea is to test, that only changes on the relevant field are reflected, probably doesn't make much sense here, and should only be tested for the mapToControlProps